### PR TITLE
Adding TOC UI

### PR DESCRIPTION
**Pull request by maykefreitas**
This is a UI propose for TOC (`[TOC]`) tag.

Proposing solution for #2294.

image

image

Boostnote TOC

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#2294: Markdown outlines](https://issuehunt.io/repos/53266139/issues/2294)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

**Comment by ZeroX-DG** [comment]
Can you make the TOC scroll-able? because right now if you have a long list of TOC, you won't be able to click the links that overflow the TOC.

**Comment by ZeroX-DG** [comment]
@maykefreitas ping

**Comment by maykefreitas** [comment]
hey @ZeroX-DG . 

Sorry for the delay. I was seek (fever) and forgot to open my GH notifications. But now I'm fine. 

For sure, It can be scrollable. I'll make a commit soon as possible. Take care :)

**Comment by maykefreitas** [comment]
hey @ZeroX-DG I just pushed 2 new commits to include scroll support. Looks good now :)

**Comment by ZeroX-DG** [comment]
Thanks! Can you delete the old markdown snapshot test file for TravisCI to re-generate it please?

**Comment by maykefreitas** [comment]
Yes, for sure. I forgot it

Em 28 de nov de 2019 23:27, Nguyen Viet Hung <notifications@github.com> escreveu:

Thanks! Can you delete the old markdown snapshot test file for TravisCI to re-generate it please?

**Comment by maykefreitas** [comment]
@ZeroX-DG 
done

File: tests/lib/snapshots/markdown-test.js.md [hunk 1]
```diff
@@ -226,39 +226,4 @@ Generated by [AVA](https://ava.li).
 > Snapshot 2
 
     `<p data-line="0">This is a &quot;QUOTE&quot;.</p>␊
-    
-
-## Markdown.render() should render PlantUML Ditaa correctly
```
Comment (ZeroX-DG) [inline tests/lib/snapshots/markdown-test.js.md]: Why do you have to remove these tests?

Comment (maykefreitas) [reply tests/lib/snapshots/markdown-test.js.md]: I just deleted the snapshots, to regenerate it. Do I need to it in another way?

Comment (ZeroX-DG) [reply tests/lib/snapshots/markdown-test.js.md]: That's strange, somehow it's still pass the test

Comment (ZeroX-DG) [reply tests/lib/snapshots/markdown-test.js.md]: How did you deleted them?

Comment (maykefreitas) [reply tests/lib/snapshots/markdown-test.js.md]: a normal delete, direct from the IDE

Comment (ZeroX-DG) [reply tests/lib/snapshots/markdown-test.js.md]: Oh they have been moved to the top of the file :smile: look good to me then :smile:

Comment (maykefreitas) [reply tests/lib/snapshots/markdown-test.js.md]: heheh troll

**Pull request merged**